src/figuresClasses.ts

```diff
@@ -1,19 +1,100 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.isTriangle() === false) {
+      throw new Error(`Sides ${this.a}, ${this.b}
+        and ${this.c} can't form a triangle`);
+    }
+  }
+
+  isTriangle(): boolean {
+    return (
+      this.a + this.b > this.c
+        && this.a + this.c > this.b
+        && this.b + this.c > this.a
+        && this.a !== 0
+        && this.b !== 0
+        && this.c !== 0
+    );
+  }
 
-export class Circle {
+  getArea(): number {
+    const halfPartOfPerimeter = (this.a + this.b + this.c) / 2;
+    const triangleArea = Math.sqrt(
+      halfPartOfPerimeter * (halfPartOfPerimeter - this.a)
+      * (halfPartOfPerimeter - this.b)
+      * (halfPartOfPerimeter - this.c),
+    );
 
+    return Number(triangleArea.toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = 'circle';
+
+    if (radius <= 0) {
+      throw new Error('Is not a circle');
```

Comment:
Checklist p3
How use should understand what is wrong and what data are valid?
Fix here and in all the rest errors